### PR TITLE
 create protected routes

### LINKED ISSUE #86: [front end] create protected routes react
create protected routes react 
specify the responsibility and availability of our pages

### PR DESCRIPTION
**Pull request by mo7amedehab97**
create protected routes

- create the dashboard link if the user admin 
- add the navbar to the dashboard page 
- add navbar to the product pages 
- create the protected routes 
- related #86

### CODE REVIEW

File: client/src/Component/Navbar/index.jsx [hunk 1]
```diff
@@ -26,8 +26,9 @@ function Navbar() {
       </section>
       <section className="navbar-info">
         <Link to="/">Home</Link>
-        <Link to="/shop">Shop </Link>
-        <Link to="/services">Services</Link>
+        <Link to="/products">Shop </Link>
+        <Link to="/eng">Engineer</Link>
+        {user && <Link to="/dashboard">Dashboard</Link>}
```
Comment (aaamra) [inline client/src/Component/Navbar/index.jsx]: here the user should be an admin to see the dashboard link
so check the admin role not just the user

File: client/src/ProtectedRoutes/ProtectedRoutes.jsx [hunk 2]
```diff
@@ -11,25 +12,26 @@ import {
...
-    return <Navigate to="/dashboard" replace />;
+  if (user?.role !== ADMIN_ROLE) {
+    return <Navigate to="/" replace />;
   }
-  return children;
+  return children || <Outlet />;
 }
 export function ProtectedUser({ children }) {
   const { user } = useAuth();
   if (user) {
```
Comment (aaamra) [inline client/src/ProtectedRoutes/ProtectedRoutes.jsx]: if there is no user you should navigate to home page
```suggestion
  if (!user) {
```

**Review by aaamra — CHANGES_REQUESTED** [review]

File: client/src/Component/Navbar/index.jsx (hunk 1 above)
Comment (mo7amedehab97) [reply client/src/Component/Navbar/index.jsx]: done

File: client/src/ProtectedRoutes/ProtectedRoutes.jsx (hunk 2 above)
Comment (mo7amedehab97) [reply client/src/ProtectedRoutes/ProtectedRoutes.jsx]: done

**Review by nabilramy — DISMISSED** [review]

**Review by aaamra — DISMISSED** [review]

**Review by aaamra — APPROVED** [review]

**Review by nabilramy — APPROVED** [review]

**Review by braaAwni — APPROVED** [review]

**Review by KAHMOOSHA — APPROVED** [review]

**Pull request merged**